lib/features/onboarding/OnBoardingScreen.dart

```diff
@@ -0,0 +1,204 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:rick_morty_universe/features/authentication/AuthenticationScreen.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'OnBoardingScreen.g.dart';
+
+@riverpod
+class OnBoardingProgress extends _$OnBoardingProgress {
+  @override
+  double build() {
+    return 0.25;
+  }
+
+  void updateProgress(int pageIndex) {
+    state = pageIndex == 0
+        ? 0.25
+        : pageIndex == 1
+        ? 0.50
+        : pageIndex == 2
+        ? 0.75
+        : 1;
+  }
+}
```

Comment:
It's generally a good idea to keep the state in a separate file from the UI. You can consider that file to behave like a ViewModel as it is helping you implement separation of concerns without actually having to use a ViewModel class.